### PR TITLE
doc: initial version of strategic initiatives

### PR DESCRIPTION
**Pull request by mhdawson**
I have likely missed some and included some which are not appropriate.  But a first cut to trigger discussion

@nodejs/tsc 

Related to https://github.com/nodejs/TSC/issues/390 and https://github.com/nodejs/TSC/issues/278

### CODE REVIEW

**Comment by mhdawson** [comment]
If/when we agree to land this will also add link on README.md

**Comment by MylesBorins** [comment]
The tsc readme or project readme?

If the latter should we open a parallel pr?

File: Strategic-Initiatives.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Current Intiatives
+
+| Initiative        | Champion        | Links                                                            |
+|-------------------|-----------------|------------------------------------------------------------------|
+| http2             | James Snell     |                                                                  |
+| Modules           | Myles Borins    | https://github.com/nodejs/node-eps/blob/master/002-es-modules.md |
+| N-API             | Michael Dawson  | https://github.com/nodejs/abi-stable-node                        |
+| OpenSSL Evolution | Rod Vagg        | https://github.com/nodejs/TSC/issues/364                         |
+| Workers           | Anna Henningson | https://github.com/nodejs/worker                                 |
+| Error Messages    | James Snell     | https://github.com/nodejs/node/issues/11273i                     |
```
Comment (vsemozhetbyt) [inline Strategic-Initiatives.md]: A nit: 11273i -> 11273

File: Strategic-Initiatives.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+| CVE Management    | Michael Dawson  | https://github.com/nodejs/security-wg/issues/33                  |
+| Core Promise APIs | James Snell     |                                                                  | 
+| TSC Governance    | Myles Borins    | https://github.com/nodejs/TSC/issues/383                         |
+| New Streams APIs  | James Snell     | https://github.com/ nodejs/node#16414                            |
```
Comment (vsemozhetbyt) [inline Strategic-Initiatives.md]: A nit: `https://github.com/ nodejs/node#16414` -> `https://github.com/nodejs/node/pull/16414` ?

File: Strategic-Initiatives.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+At any one time the Node.js project has a number of strategic initiatives
+underway.  The goal of the TSC is to have a `champion` for each of these
+initiatives and to support the initiatives in order to enable their
+success.
+
+A review of the initiatives will be a standing item on the TSC agenda (even
+if the update is `nothing new`) as a way to ensure they are active
+and have the support needed.
+
+# Current Intiatives
```
Comment (vsemozhetbyt) [inline Strategic-Initiatives.md]: A nit: Intiatives -> Initiatives

File: Strategic-Initiatives.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+| Modules           | Myles Borins    | https://github.com/nodejs/node-eps/blob/master/002-es-modules.md |
+| N-API             | Michael Dawson  | https://github.com/nodejs/abi-stable-node                        |
+| OpenSSL Evolution | Rod Vagg        | https://github.com/nodejs/TSC/issues/364                         |
+| Workers           | Anna Henningson | https://github.com/nodejs/worker                                 |
+| Error Messages    | James Snell     | https://github.com/nodejs/node/issues/11273i                     |
+| CVE Management    | Michael Dawson  | https://github.com/nodejs/security-wg/issues/33                  |
+| Core Promise APIs | James Snell     |                                                                  | 
+| TSC Governance    | Myles Borins    | https://github.com/nodejs/TSC/issues/383                         |
+| New Streams APIs  | James Snell     | https://github.com/ nodejs/node#16414                            | 
+| Diagnotics        | Michael Dawson  | Happy to have other volunteers but will work to restart          |
```
Comment (vsemozhetbyt) [inline Strategic-Initiatives.md]: A nit: Diagnotics    -> Diagnostics

File: Strategic-Initiatives.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# Strategic Initiatives
+
+At any one time the Node.js project has a number of strategic initiatives
+underway.  The goal of the TSC is to have a `champion` for each of these
```
Comment (mscdex) [inline Strategic-Initiatives.md]: Minor nit: I think we should avoid using backticks for things that aren't (code) keywords. Perhaps use italics, bold, or quotes instead (here and elsewhere in this document).

Comment (Trott) [reply Strategic-Initiatives.md]: ....Or use nothing. The word *champion* does not need the scare quotes here or any other typographic addition.

File: Strategic-Initiatives.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+| Workers           | Anna Henningson | https://github.com/nodejs/worker                                 |
+| Error Messages    | James Snell     | https://github.com/nodejs/node/issues/11273i                     |
+| CVE Management    | Michael Dawson  | https://github.com/nodejs/security-wg/issues/33                  |
+| Core Promise APIs | James Snell     |                                                                  | 
+| TSC Governance    | Myles Borins    | https://github.com/nodejs/TSC/issues/383                         |
+| New Streams APIs  | James Snell     | https://github.com/ nodejs/node#16414                            | 
+| Diagnotics        | Michael Dawson  | Happy to have other volunteers but will work to restart          |
+| Moderation        | Rich Trot       |                                                                  |
+| Async Hooks       | Trevor Norris   |                                                                  |
+| V8 Currency       | MichaëZasso    |                                                                  |
```
Comment (Trott) [inline Strategic-Initiatives.md]: `Michaél Zasso`

**Comment by Trott** [comment]
I don't know if they're strategic, but if so, you can put me down for collaborator onboarding and new contributor onramp.

**Comment by Trott** [comment]
Joyee for automation? Rod for build/CI?

**Comment by joyeecheung** [comment]
@Trott 
> Joyee for automation? Rod for build/CI?

Sure!

```
| Automation       | Joyee Cheung   |  https://github.com/nodejs/automation   |
```

**Comment by mhdawson** [comment]
In terms of build/automation I was not quite sure whether we should add areas covered by Working Groups.    I'm happy to add those as well if people think that makes sense.  Would include Build, Benchmarking, Release etc...

**Comment by mhdawson** [comment]
@MylesBorins I agree adding to node core readme probably also makes sense for visibility.  I'll submit a PR to do that as well once this one lands.

File: Strategic-Initiatives.md (hunk 5 above)
Comment (mhdawson) [reply Strategic-Initiatives.md]: Ok changed to remove backticks.

File: Strategic-Initiatives.md (hunk 4 above)
Comment (mhdawson) [reply Strategic-Initiatives.md]: done

File: Strategic-Initiatives.md (hunk 3 above)
Comment (mhdawson) [reply Strategic-Initiatives.md]: done

File: Strategic-Initiatives.md (hunk 2 above)
Comment (mhdawson) [reply Strategic-Initiatives.md]: done

File: Strategic-Initiatives.md (hunk 1 above)
Comment (mhdawson) [reply Strategic-Initiatives.md]: done

File: Strategic-Initiatives.md (hunk 6 above)
Comment (mhdawson) [reply Strategic-Initiatives.md]: Will try, the special characters always mess me up.

File: Strategic-Initiatives.md (hunk 1 above)
Comment (vsemozhetbyt) [reply Strategic-Initiatives.md]: Still not?

File: Strategic-Initiatives.md (hunk 2 above)
Comment (vsemozhetbyt) [reply Strategic-Initiatives.md]: Space is fixed, but the URL is still wrong?

File: Strategic-Initiatives.md (hunk 3 above)
Comment (vsemozhetbyt) [reply Strategic-Initiatives.md]: Still not?

File: Strategic-Initiatives.md (hunk 4 above)
Comment (vsemozhetbyt) [reply Strategic-Initiatives.md]: Still not?

**Comment by mhdawson** [comment]
@Trott  any chance you can push a commit to fix M Zasso's name.  I tried my usual tricks of copying from other docs but did not work...

**Comment by mhdawson** [comment]
Went ahead an added the WG's as well so people can see how it looks

**Comment by mhdawson** [comment]
@vsemozhetbyt @mscdex @Trott pushed commits to address comments.

File: Strategic-Initiatives.md (hunk 6 above)
Comment (vsemozhetbyt) [reply Strategic-Initiatives.md]: Still there)

**Comment by mhdawson** [comment]
@MylesBorins added commit to add reference in README.md for TSC.  Once this lands will open PR to do the same in the core node repo.

**Comment by MylesBorins** [comment]
Great work on this @mhdawson 

I'd like to suggest that we keep the scope of initiatives to projects as opposed to working groups themselves. For example, James' work on a new lower level streams implementation is a much tighter scope than streams wg.

I would also like to suggest that we keep the number of initiatives that we are working on small and encourage individuals to add their initiatives. If people are not openly taking responsibility for an initiative in this thread it likely should not be included.

There are likely opportunities for the champion to not be a member of the TSC, we might want to account for that. A redesign of the nodejs.org website may be an example of that.

I've marked this for the agenda, we should discuss it on Wednesday

**Comment by mhdawson** [comment]
I'm in agreement that it is best to omit WG's for now.  Will remove and we can disuss further in the meeting

**Comment by mhdawson** [comment]
I'd proposed that people suggest what they think the list should be (or specific additions/removals from the list above).  In the absence of any other concrete suggestions we could just start with the current list so that we don't stall out for too long as there has not been much discussion over the last week (I do know people have been busy though)

**Comment by Fishrock123** [comment]
I would suggest swapping @jasnell on the Streams API work for myself. He's already got a lot of things to champion and I both don't mind taking this one and have kinda been doing that role to begin with-ish?

**Comment by Fishrock123** [comment]
Well?

File: Strategic-Initiatives.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+| http2             | James Snell     |                                                                  |
+| Modules           | Myles Borins    | https://github.com/nodejs/node-eps/blob/master/002-es-modules.md |
+| N-API             | Michael Dawson  | https://github.com/nodejs/abi-stable-node                        |
+| OpenSSL Evolution | Rod Vagg        | https://github.com/nodejs/TSC/issues/364                         |
+| Workers           | Anna Henningson | https://github.com/nodejs/worker                                 |
+| Error Messages    | James Snell     | https://github.com/nodejs/node/issues/11273                      |
+| CVE Management    | Michael Dawson  | https://github.com/nodejs/security-wg/issues/33                  |
+| Core Promise APIs | James Snell     |                                                                  | 
+| TSC Governance    | Myles Borins    | https://github.com/nodejs/TSC/issues/383                         |
+| New Streams APIs  | James Snell     | https://github.com/nodejs/node/pull/16414                        |
```
Comment (fhinkel) [inline Strategic-Initiatives.md]: I think this should be @Fishrock123 instead of @jasnell?

**Comment by fhinkel** [comment]
LGTM with the change for Streams API. 

Should this doc include a list of past initiatives? If so, I'd like to add myself for fixing the vm module 😊

**Comment by MylesBorins** [comment]
I think we should drop diagnostics + automation. If there are specific initiatives within those groups (such as a commit queue or specific diagnostics efforts) those would make sense to include

Collab onboarding + collab on ramp seem like a single initiative to me, thoughts @Trott

**Comment by Trott** [comment]
> Collab onboarding + collab on ramp seem like a single initiative to me, thoughts @Trott

Sure. We can always split them up later if we need to.

**Comment by Fishrock123** [comment]
Removing from the agenda as there is no progress to discuss.

File: Strategic-Initiatives.md (hunk 7 above)
Comment (jasnell) [reply Strategic-Initiatives.md]: @Fishrock123 *and* @jasnell ... we're both working two different sides of this.

**Comment by Fishrock123** [comment]
Fwiw, we went over these live and got updates on each in the TSC meeting for https://github.com/nodejs/TSC/issues/418 which can be found in the last half of https://youtu.be/eSgqJg7pu_M

**Comment by mhdawson** [comment]
Pushed update to address comments.  @fhinkel can you give me a link to add for the vm link.

@Fishrock123 can you give me a link to add for the streams 2? From what I can tell from the discussion they are really 2 different initiatives so listing them separately is what I was thinking makes sense.

Once I get those 2 I'll land and we can iterate as we move forward.

**Comment by fhinkel** [comment]
@mhdawson Here's the link to the commit that fixed most of the vm issues: https://github.com/nodejs/TSC/pull/399

Can you add a current initiative, please: 
Initiative: C++ smart pointers
Champion: Franziska Hinkelmann
Links: https://github.com/nodejs/node/blob/master/CPP_STYLE_GUIDE.md#ownership-and-smart-pointers

**Comment by jasnell** [comment]
@mhdawson ... re: streams, they are not two different initiatives... there are just two sides to one initiative.. that is, reworking streams on both the C++ and JS side.

**Comment by mhdawson** [comment]
@fhinkel the the first link seems to be for this PR as opposed to the vm issues.

**Comment by mhdawson** [comment]
@fhinkel is the initiative to move to using smart pointers or something else ?

**Comment by mhdawson** [comment]
@fhinkel I hope you don't mind but I'm going to go ahead and land, can you submit a Pr to add the new initiative and add the missing link for the past vm initiative ?  Alternatively if you send me that info I can open a new PR to update for that.

**Comment by mhdawson** [comment]
Landed as cf87134911d34312d4dc8a52362bad939a339905

**Comment by mhdawson** [comment]
See I missed removing one line, opening a PR to fix.

**Pull request closed without merging**